pkg/scheduler/schedule_one.go

```diff
@@ -413,7 +413,7 @@ func (sched *Scheduler) findNodesThatFitPod(ctx context.Context, fwk framework.F
 			nodes = append(nodes, nInfo)
 		}
 	}
-	feasibleNodes, err := sched.findNodesThatPassFilters(ctx, fwk, state, pod, diagnosis, nodes)
+	feasibleNodes, err := sched.findNodesThatPassFilters(ctx, fwk, state, pod, diagnosis, nodes, true)
```

Comment:
instead of adding a flag, lets just update `nextStartNodeIndex` here:

```
processedNodes := len(feasibleNodes) + len(diagnosis.NodeToStatusMap)
sched.nextStartNodeIndex = (sched.nextStartNodeIndex + processedNodes) % len(nodes)
```

Reply:
@ahg-g Thanks for you reply.
But this probably didn't fit our previous design. We always update the `nextStartNodeIndex` before, whether there is an error or not. This ensures that all nodes have a chance to be searched. If we leave the update logic outside, it will be difficult to update the `nextStartNodeIndex` when an error was returned by `findNodesThatPassFilters`.  Because we don't know how many nodes we've processed in `findNodesThatPassFilters`.  Changing the return value type of `findNodesThatPassFilters` function may not seem like a good idea. 
https://github.com/kubernetes/kubernetes/blob/47ad357c52713aa692aff3b6906e16f27231bab1/pkg/scheduler/schedule_one.go#L516-L519

Reply:
> simply return the feasibleNodes on error at line 524

@ahg-g thanks, it seems pretty clear. I have updated the code.